pkg/operation/botanist/component/logging/kuberbacproxy/kube_rbac_proxy_test.go

```diff
@@ -0,0 +1,259 @@
+			Expect(managedResourceSecret.Type).To(Equal(corev1.SecretTypeOpaque))
+			Expect(managedResourceSecret.Data).To(HaveLen(3))
+
+			Expect(string(managedResourceSecret.Data["clusterrole____gardener.cloud_logging_promtail.yaml"])).To(Equal(serialize(&rbacv1.ClusterRole{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: rbacv1.SchemeGroupVersion.String(),
+					Kind:       "ClusterRole",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "gardener.cloud:logging:promtail",
+					Labels: map[string]string{
+						"app": promtailName,
+					},
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{
+							"",
+						},
+						Resources: []string{
+							"nodes",
+							"nodes/proxy",
+							"services",
+							"endpoints",
+							"pods",
+						},
+						Verbs: []string{
+							"get",
+							"list",
+							"watch",
+						},
+					},
+					{
+						NonResourceURLs: []string{
+							"/loki/api/v1/push",
+						},
+						Verbs: []string{
+							"create",
+						},
+					},
+				},
+			})))
+			Expect(string(managedResourceSecret.Data["clusterrolebinding____gardener.cloud_logging_kube-rbac-proxy.yaml"])).To(Equal(serialize(&rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "gardener.cloud:logging:kube-rbac-proxy",
+					Labels: map[string]string{
+						"app": kubeRBACProxyName,
+					},
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.GroupName,
+					Kind:     "ClusterRole",
+					Name:     "system:auth-delegator",
+				},
+				Subjects: []rbacv1.Subject{{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      kubeRBACProxyName,
+					Namespace: metav1.NamespaceSystem,
+				}},
+			})))
+			Expect(string(managedResourceSecret.Data["clusterrolebinding____gardener.cloud_logging_promtail.yaml"])).To(Equal(serialize(&rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "gardener.cloud:logging:promtail",
+					Labels: map[string]string{
+						"app": promtailName,
+					},
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.GroupName,
+					Kind:     "ClusterRole",
+					Name:     "gardener.cloud:logging:promtail",
+				},
+				Subjects: []rbacv1.Subject{{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      promtailName,
+					Namespace: metav1.NamespaceSystem,
+				}},
+			})))
+		})
+	})
+
+	Describe("#Destroy", func() {
+		It("should successfully destroy all resources", func() {
+			Expect(c.Create(ctx, managedResource)).To(Succeed())
+			Expect(c.Create(ctx, managedResourceSecret)).To(Succeed())
+
+			Expect(kubeRBACProxyDeployer.Destroy(ctx)).To(Succeed())
+
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(BeNotFoundError())
+			Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(BeNotFoundError())
+		})
+	})
+})
+
+func serialize(obj client.Object) string {
```

Comment:
Hmm, I see that serialize is now used in more places than I thought initially. Does it makes sense to extract in e.g. `pkg/operation/botanist/component/test/seralize.go`?